Add USB converter for AT&T 56K series keyboards (#8086)
qmk/qmk_firmware: keyboards/converter/att56k_usb/README.md

# AT&T 56K 122 key terminal keyboard converter

A USB converter for AT&T 56K series keyboards.
* Keyboard Maintainer: [engicoder](https://github.com/engicoder)
* Hardware Supported: 
  - Converter: Pro Micro 5V, Teensy 2.0, ATmega32u4
  - Keyboards: 
    - AT&T 56K 460 ACW - 122 key magnetic separation
    - AT&T 56K 560 AGA - 122 key buckling spring

## Interfacing

The 56k keyboards use a 16 bit data packet. Depending on the keyboard the 

> transmission maybe synchronous with a 9.6Kz clock frequency or ansynchronous

Reviewer: ```suggestion
transmission may be synchronous with a 9.6kHz clock frequency or asynchronous
```

### Data Frame
The data frame is 16 bits of which the top 8 bits are always 0x7F.
The lower 8 bits contain the scan code. MAKE or BREAK is indicated in the 
most significant bit (bit 7), a value of 1 indicating a MAKE
```
 |            Header             |           Scan Code           |
 +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
 |              0x7F             | M | 6 | 5 | 4 | 3 | 2 | 1 | 0 |
 +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
 ```

### Timing Diagram
Timing diagram for the press (MAKE) of the A key which has a scan code of 0x35.
```
     _____   _   _   _   _   _   _   _   _   _   _   _   _   _   _   _   _____________
CLOCK     \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/
     ____    _______________________________     _______     ___     _________________
DATA     \__/                               \___/       \___/   \___/
     _____________________________________________________________________          __
Click                                                                     \________/
```
DATA transitions low half a clock cycle before CLOCK to ensure the leading 0 bit value. 
## Build

> From the root qmk_firmware directory execute:

Reviewer 2: ```suggestion
Make example for this keyboard (after setting up your build environment):
```

>     $ make converter/att56k_usb

Reviewer 2: ```suggestion
    make converter/att56k_usb:default
```

> 

Reviewer 2: ```suggestion
```

> To specify a custom keymap in the folder <custom_keymap>, execute

Reviewer 2: ```suggestion
```

> 

Reviewer 2: ```suggestion
```

>     $ make converter/att56k_usb:custom_keymap

Reviewer 2: ```suggestion
```
